File: lab7_1/java/Message.java
package lab7;

public class Message {


    private final Position endPosition;
    private final String message;

    public Message(Position endPosition, String message) {
        this.endPosition = endPosition;
        this.message = message;
    }

    @Override
    public String toString() {
        return "ERROR " + this.endPosition + ": " + this.message;
    }
}

File: lab7_1/lexer_ts/src/Message.ts
import { Position } from "./Position";

class Message {


    private endPosition: Position;
    private message: string;

    constructor(endPosition: Position, message: string) {
        this.endPosition = endPosition;
        this.message = message;
    }

    public toString(): string {
        return 'ERROR ' + this.endPosition.toString() + ': ' + this.message;
    }
}

export { Message }
